sdstore/sdstore.go

```diff
@@ -349,25 +325,12 @@ func (s *sdStore) putFile(url *url.URL, bodyType string, filePath string) error
 	}
 	fsize := stat.Size()
 
-	reader, writer := io.Pipe()
-
-	done := make(chan error)
-	go func() {
-		_, err := s.put(url, bodyType, reader, fsize)
-		if err != nil {
-			done <- err
-			return
-		}
-
-		done <- nil
-	}()
-
-	io.Copy(writer, input)
-	if err := writer.Close(); err != nil {
+	_, err = s.put(url, bodyType, input, fsize)
```

Comment:
Is it better to use `io.Pipe`? I try not to use the function. If there are problem, please tell me.